src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -455,7 +455,14 @@ impl<'tcx> LinkCollector<'_, 'tcx> {
                 )
             })
             .and_then(|res| res.try_into().ok())
-            .or_else(|| resolve_primitive(path_str, ns));
+            .or_else(|| resolve_primitive(path_str, ns))
+            .or_else(|| {
+                if self.cx.tcx.crate_name(LOCAL_CRATE).as_str() == path_str {
+                    Some(Res::from_def_id(self.cx.tcx, CRATE_DEF_ID.to_def_id()))
+                } else {
+                    None
+                }
+            });
```

Comment:
I'm not sure if this is the right place to put it. This runs in the context of the crate that's being documented, not in the context of the item having its names resolved, which seems like it would behave wrong when cross-crate inlining happens.

Shouldn't the `doc_link_resolutions` (in `rustc_resolve`) query be updated instead?

Reply:
Hum, fair enough. Gonna move it there then.